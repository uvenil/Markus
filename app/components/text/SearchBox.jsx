'use strict';

import React from 'react';
import Unique from '../../utils/Unique';
import Config from '../../../config.json';
import _ from 'lodash';

export default class SearchBox extends React.Component {
    constructor(props) {
        super(props);

        this._textBoxId     = Unique.elementId('a');
        this._clearButtonId = Unique.elementId('b');

        this._handleChange = value => {
            document.getElementById(this._clearButtonId).style.display = _.isEmpty(value) ? 'none' : 'block';

            if (this.props.onChange) {
                this.props.onChange(value);
            }
        };

        this._handleClear = () => {
            const textBox = document.getElementById(this._textBoxId);
            textBox.value = '';
            textBox.focus();

            this._handleChange('');
        };
    }

    render() {
        const theme     = this.props.theme === 'dark' ? require('../../theme.dark.json') : require('../../theme.light.json');
        const textColor = this.props.textColor ? this.props.textColor : theme.primaryTextColor;

        return (
            <div
                className={this.props.className}
                style={{ width : '100%', position : 'relative', pointerEvents : this.props.disabled ? 'none' : 'auto' }}>
                <i
                    className="fa fa-fw fa-search"
                    style={{ position : 'absolute', left : Config.paddingX1 + 'px', top : Config.paddingX0 + 'px', fontSize : this.props.textSize, color : '#888', cursor : 'default' }} />
                <input
                    id={this._textBoxId}
                    type="text"
                    placeholder={this.props.hintText}
                    style={{ width : 'calc(100% - 4 * ' + Config.paddingX2 + 'px)', WebkitUserSelect : 'none', fontFamily : this.props.fontFamily, fontSize : this.props.textSize, color : textColor, backgroundColor : theme.secondaryBackgroundColor, outline : 'none', borderWidth : '1px', borderStyle : 'solid', borderRadius : Config.paddingX2 + 'px', borderColor : theme.borderColor, paddingLeft : 'calc(2 * ' + Config.paddingX2 + 'px)', paddingRight : 'calc(2 * ' + Config.paddingX2 + 'px)', paddingTop : Config.paddingX0 + 'px', paddingBottom : Config.paddingX0 + 'px' }}
                    onChange={event => this._handleChange(event.target.value)} />
                <i
                    id={this._clearButtonId}
                    className="fa fa-fw fa-times-circle"
                    style={{ display : 'none', position : 'absolute', right : Config.paddingX1 + 'px', top : Config.paddingX0 + 'px', color : '#888', cursor : 'pointer' }}
                    onClick={this._handleClear} />
            </div>
        );
    }
}

SearchBox.propTypes = {
    hintText   : React.PropTypes.string,
    className  : React.PropTypes.string,
    fontFamily : React.PropTypes.string,
    textSize   : React.PropTypes.string,
    textColor  : React.PropTypes.string,
    theme      : React.PropTypes.oneOf([ 'light', 'dark' ]),
    disabled   : React.PropTypes.bool,
    onChange   : React.PropTypes.func
};

SearchBox.defaultProps = {
    hintText  : 'Search',
    className : 'SearchBox',
    theme     : 'light',
    disabled  : false
};

module.exports = SearchBox;
