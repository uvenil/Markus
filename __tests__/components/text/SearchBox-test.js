'use strict';

import React from 'react';
import Renderer from 'react-test-renderer';
//import { shallow } from 'enzyme';
import SearchBox from '../../../app/components/text/SearchBox.jsx';

jest.mock('../../../app/utils/Unique');

it('renders correctly', () => {
    const searchBox = Renderer.create(
        <SearchBox />
    ).toJSON();

    expect(searchBox).toMatchSnapshot();
});

/*it('shows clear button after text input', () => {
    const searchBox = shallow(
        <SearchBox />
    );

    const input       = searchBox.find('input');
    const clearButton = searchBox.find('i')[1];

    expect(input.value).toEqual('');
    expect(clearButton.style.display).toEqual('none');

    input.simulate('keyPress', { which : 'a' });

    expect(input.value).toEqual('a');
    expect(clearButton.style.display).toEqual('block');
});*/
