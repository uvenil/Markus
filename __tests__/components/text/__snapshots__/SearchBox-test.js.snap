exports[`test renders correctly 1`] = `
<div
  className="SearchBox"
  style={
    Object {
      "pointerEvents": "auto",
      "position": "relative",
      "width": "100%",
    }
  }>
  <i
    className="fa fa-fw fa-search"
    style={
      Object {
        "color": "#888",
        "cursor": "default",
        "fontSize": undefined,
        "left": "8px",
        "position": "absolute",
        "top": "4px",
      }
    } />
  <input
    id="a"
    onChange={[Function]}
    placeholder="Search"
    style={
      Object {
        "WebkitUserSelect": "none",
        "borderColor": "#bfbfbf",
        "borderRadius": "16px",
        "borderStyle": "solid",
        "borderWidth": "1px",
        "color": "#000",
        "fontFamily": undefined,
        "fontSize": undefined,
        "outline": "none",
        "paddingBottom": "4px",
        "paddingLeft": "calc(2 * 16px)",
        "paddingRight": "calc(2 * 16px)",
        "paddingTop": "4px",
        "width": "calc(100% - 4 * 16px)",
      }
    }
    type="text" />
  <i
    className="fa fa-fw fa-times-circle"
    id="b"
    onClick={[Function]}
    style={
      Object {
        "color": "#888",
        "cursor": "pointer",
        "display": "none",
        "position": "absolute",
        "right": "8px",
        "top": "4px",
      }
    } />
</div>
`;
